libraries/Microsoft.Bot.Builder.Skills/Dialogs/SkillDialog.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Bot.Builder.Skills.Dialogs;
+using Microsoft.Bot.Builder.TraceExtensions;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Dialogs
+{
+    /// <summary>
+    /// A sample dialog that can wrap remote calls to a skill.
+    /// </summary>
+    /// <remarks>
+    /// The options parameter in <see cref="BeginDialogAsync"/> must be a <see cref="SkillDialogArgs"/> instance
+    /// with the initial parameters for the dialog.
+    /// </remarks>
+    public class SkillDialog : Dialog
+    {
+        private readonly IStatePropertyAccessor<BotFrameworkSkill> _activeSkillProperty;
+        private readonly ConversationState _conversationState;
+        private readonly SkillDialogOptions _dialogOptions;
+
+        public SkillDialog(SkillDialogOptions dialogOptions, ConversationState conversationState)
```

Comment:
>SkillDialogOptions dialogOptions,  [](start = 27, length = 34)

I would just make these properties.

Reply:
OK, I'll move them to properties but this is something I don't like much from adaptive, having them as constructor parameters or explicit parameters in BeginConversationAsync communicates better that those are one time only properties and can't be altered once the dialog is created or initiated. Haven them as properties makes the programming model more brittle.